### PR TITLE
Towards CMake-based OS X kernel extension building

### PR DESCRIPTION
**Pull request by theopolis**
This PR includes a rebase on facebook/master, so it includes the previous 2 kernel-feature commits.

### CODE REVIEW

**Comment by theopolis** [comment]
This is essentially: https://github.com/facebook/osquery/commit/573982aa62be1950c60fe8b5e714e79015bf6b15

File: docs/wiki/development/kernel.md [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
-Once kernel extension signature verification is disabled, just `make` the kernel extension.  It can then be loaded and unloaded with `make load` and `make unload`.  For permanent installations it should be added to the systems startup kernel extensions.
+The high-level kernel development workflow involves:
+- Install a copy of OS X in VMware and share a development/source/build folder.
+- Compile and build the user-land and kernel-land code on your host.
+- Load the kernel extension in the VM and run the kernel-specific unit tests.
+- You may optionally configure the VM to break on boot to debug with `lldb`.
+
+## Building on OS X for local loading (not recommended)
+
+First disable signature verification on the machine by running `make kernel-provision` a reboot is required for this to take effect.  **CAUTION:** this lowers the security of your system.  It is better to have your kernel extensions signed on your host and develop/load in a virtual machine.
```
Comment (theopolis) [inline docs/wiki/development/kernel.md]: We'll change this to `make kernel-deps`.

File: docs/wiki/development/kernel.md [hunk 2]
```diff
@@ -1,28 +1,47 @@
...
+
+```
+make kernel-build
+make kernel-load
+# Optionally run the unit tests
+# make kernel-test
```
Comment (theopolis) [inline docs/wiki/development/kernel.md]: `kernel-test` will require sudo and will assume the kernel module/extension is loaded.

**Comment by theopolis** [comment]
Update: rebased on kernel branch with end-to-end/integration test fixes. ok to test

**Comment by theopolis** [comment]
Now we will see correct failures since the kernel build is OS X only, but is not restricted by platform yet.

**Comment by theopolis** [comment]
ok to test

**Comment by marpaia** [comment]
This is going to be boss. CMake so hard.

**Comment by theopolis** [comment]
ok to test

**Pull request merged**